docs/design/node-group.md

```diff
@@ -0,0 +1,50 @@
+## Introduction
+
+nodegroup is the nodes with the same specified label, on the basis of nodegroup, different queues and nodegroups have some relationship about affinity and anti-affinity.
+
+case1: different departments (nlp and recommend) use different nodes (nodegroup1 and nodegroup2), tts uses nodegroup1 first, and nodegroup2 can also be used when resources are insufficient, but recommend can only use nodegroup2.
+
+![](images/node-group-case-1.png)
+
+case2: recommend can use private cloud nodes or public cloud nodes, but tts can only use private cloud nodes because its task takes a long time, which is not easy to the release of public cloud nodes (for saving money)
+
+![](images/node-group-case-2.png)
+
+## Solution
+
+1. First, we need mark out some nodes(by add specific label for nodes) which are the same group.
+2. Second, we need to express the relationship between queue and nodegroup, such as affinity and antiaffinity
+   1. groupaffinity.required, hard constraints, such as `nlp = nodegroup1,nodegroup2`, it means that task in queue=nlp can ony run on the nodes in nodegroup1 or nodegroup2.
+   2. groupaffinity.preferred, soft constraints, such as `nlp = nodegroup1`, it means that task in queue=nlp runs on nodegroup1 first, but if the resources of nodegroup1 is insufficient, it can also run on other nodegroups.
+   3. groupantiaffinity.required, hard constraints, such as `nlp = nodegroup1`, it means that task in queue=nlp can run on any nodegroups but nodegroup1.
+   4. groupantiaffinity.preferred, soft constraints, such as `nlp = nodegroup1`, it means that task in queue=nlp runs on any other nodegroups, but if the resources of other nodegroup1 is insufficient, it can also run on nodegroup1.
+
+example
+
+```yaml
+actions: "reclaim, allocate, backfill, preempt"
+tiers:
+- plugins:
+  - name: priority
+  - name: gang
+  - name: conformance
+- plugins:
+  - name: drf
+  - name: predicates
+  - name: proportion
+  - name: nodegroup
```

Comment:
As it's a common scenaro. Let's discuss to enhance queue to support it insted of a plugin.